src/sync/impl/sync_metadata.cpp

```diff
@@ -426,6 +452,54 @@ std::shared_ptr<Realm> SyncMetadataManager::get_realm() const
     return realm;
 }
 
+/// Magic key to fetch app metadata, which there should always only be one of.
+static const auto app_metadata_pk = 1;
+
+void SyncMetadataManager::set_app_metadata(const std::string& deployment_model,
+                                           const std::string& location,
+                                           const std::string& hostname,
+                                           const std::string& ws_hostname) const
+{
+    if (m_app_metadata) {
+        return;
+    }
+    
+    auto realm = get_realm();
+    auto& schema = m_app_metadata_schema;
+    
+    realm->begin_transaction();
+    
+    auto table = ObjectStore::table_for_object_type(realm->read_group(), c_sync_app_metadata);
+    auto obj = table->create_object_with_primary_key(app_metadata_pk);
+    obj.set(schema.idx_deployment_model, deployment_model);
+    obj.set(schema.idx_location, location);
+    obj.set(schema.idx_hostname, hostname);
+    obj.set(schema.idx_ws_hostname, ws_hostname);
+
+    realm->commit_transaction();
```

Comment:
What if the transaction fails?

Reply:
Then the `/location` endpoint will be hit for each network call. I'm not sure this can be fixed properly until we have the monorepo.